src/main/java/com/minecraftabnormals/abnormals_core/core/mixin/EntityMixin.java

```diff
@@ -131,10 +133,11 @@ public void read(CompoundNBT compound, CallbackInfo info) {
 		}
 	}
 
-	@Redirect(method = "move", at = @At(value = "INVOKE", target = "Lnet/minecraft/block/Block;stepOn(Lnet/minecraft/world/World;Lnet/minecraft/util/math/BlockPos;Lnet/minecraft/entity/Entity;)V"))
-	private void onEntityWalk(Block block, World world, BlockPos pos, Entity entity) {
-		if (!EntityWalkEvent.onEntityWalk(world, pos, entity)) {
-			block.stepOn(world, pos, entity);
+	@Redirect(method = "move", at = @At(value = "INVOKE", target = "Lnet/minecraft/entity/Entity;isSteppingCarefully()Z"))
+	private boolean checkIsSteppingCarefullyAndCanEntityWalkOn(Entity instance) {
```

Comment:
This is a pretty verbose name.
```suggestion
	private boolean onIsSteppingCarefully(Entity instance) {
```